src/http/buf.rs

```diff
@@ -226,6 +227,20 @@ impl MemSlice {
     fn get(&self) -> &[u8] {
         unsafe { &(*self.buf.get())[self.start..self.end] }
     }
+
+    pub fn as_str(&self) -> &str {
```

Comment:
I'm not so sure to understand the usefulness of having a new struct for this. Just making the method unsafe should be enough, no?